### PR TITLE
feat: support package mode

### PR DESCRIPTION
**Pull request by cavvia**
[PHIRE-1317](https://artsyproduct.atlassian.net/browse/PHIRE-1317)

We need to include Fortress as a packaged dependency in [Zenith](https://github.com/artsy/zenith/) so we can talk to Vault as part of our CI flow. The `package-mode = false` flag makes it impossible to build it as a dependency. Is there a reason it has been added? I know we run it as a container with an init script in other contexts... was its status as a package interfering with that? If it's an issue to convert into a package we could maintain a package release branch separately perhaps, but it does create complexity. I will try to look into packaging this such that we can still run the script entrypoint without modification, since it is referred to in many projects.

[PHIRE-1317]: https://artsyproduct.atlassian.net/browse/PHIRE-1317?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by artsyjian** [comment]
I vaguely recall running into problem with `setuptools` and resolved it by that flag. No, there was no intention to block package mode. I'll test running the script in this branch.

**Comment by cavvia** [comment]
@artsyjian We need to move the lib folder into a package namespace (`src/fortress`) and then ensure that the project gets packaged prior to running the `src/load/load.py` entrypoint. I think this is just a case of running `pip install .` in the root folder when building the image. I've updated the PR for the new layout and I can now load the dependency in Zenith.

**Comment by cavvia** [comment]
@artsyjian I now have `make test` working as well as the load entrypoint, but only under the poetry environment, so you have to run:

```
poetry run python src/load/load.py
```

For the `fortress` package to be loadable from the script. Other deps such as `hvac` have been moved into the package as they will not be available in the script context.

Note that `make test` now runs:

```
poetry install --with test && poetry run pytest src
```

I'm actually not sure how to get the raw script endpoint working without having it issue a bash command which wraps another script with a `poetry run`, so I may test that next.

**Comment by artsyjian** [comment]
> In our other projects we generally use pip with setuptools for packaging rather than poetry

i am fine with switching to Pip here.

**Comment by cavvia** [comment]
@artsyjian I was able to fix this by bringing back the `poetry config virtualenvs.create false` declaration in the docker image. That installs deps using the system python into shared system env, giving the test phase and the script entrypoint access to the package dependencies. Meanwhile the `fortress` package is now namespaced when loading it as a dependency into other projects. So I think this is good to go. 

Now running:

```
ENV=development python src/load/load.py iam staging hokusai-sandbox
```

Gives the desired output in `secrets.txt` both in the project root and in the docker container, as long as env vars are present.

**Comment by artsyjian** [comment]
> So I think this is good to go.

nice!

> Perhaps we can do an integration test on staging with a kubernetes deployment as a next step?

yeah, i'll test using [hokusai-sandbox](https://github.com/artsy/hokusai-sandbox) app.

File: src/load/load.py [hunk 1]
```diff
@@ -81,6 +82,41 @@ class VaultArgs:
     kvv2_mount_point: str
 
 
+def load_secrets(vault_args, project, sa_token_path, secrets_file):
+    """load secrets from  Vault and write them to a file"""
+    logging.debug(
+        f"Loading secrets from {vault_args.host}:{vault_args.port}"
+        f"under {vault_args.kvv2_mount_point}"
```
Comment (artsyjian) [inline src/load/load.py:89]: nice. didn't know lines can be broken down this way. i'll add a space after the first line, as:
```
DEBUG:root:Loading secrets from vault.stg.artsy.systems:8300under kvv2
```

File: src/load/load.py [hunk 2]
```diff
@@ -50,6 +49,8 @@ def parse_env():
     """parse env vars"""
     # load env files for local dev
     if ENV == "development":
+        from dotenv import load_dotenv
```
Comment (artsyjian) [inline src/load/load.py:52]: oh wow, conditional import, neat.

**Comment by artsyjian** [comment]
kubernetes integration works:
```
# push fortress test image
(fortress) ip-10-0-0-17:fortress jxu$ hokusai registry push --tag test --force --overwrite

# point hokusai-sandbox staging to fortress test image
ip-10-0-0-17:hokusai-sandbox jxu$ git diff
diff --git a/hokusai/staging.yml b/hokusai/staging.yml
index 2fbab7c..a9412c0 100644
--- a/hokusai/staging.yml
+++ b/hokusai/staging.yml
@@ -32,7 +32,7 @@ spec:
     spec:
       initContainers:
       - name: setenv
-        image: 585031190124.dkr.ecr.us-east-1.amazonaws.com/fortress:staging
+        image: 585031190124.dkr.ecr.us-east-1.amazonaws.com/fortress:test
         imagePullPolicy: Always
         command:
         - python

# update hokusai-sandbox deployment
ip-10-0-0-17:hokusai-sandbox jxu$ hokusai staging update --skip-checks

# show hokusai-sandbox pods
ip-10-0-0-17:hokusai-sandbox jxu$ hokusai staging status

# check secrets file in live hokusai-sandbox pod
artsy-2:~ jxu$ kubectl exec -it hokusai-sandbox-web-68995b7f5b-xjw4c cat /secrets/secrets
kubectl exec [POD] [COMMAND] is DEPRECATED and will be removed in a future version. Use kubectl exec [POD] -- [COMMAND] instead.
Defaulted container "hokusai-sandbox-web" out of: hokusai-sandbox-web, setenv (init)
export DATABASE_URL='bar'
export FOO='baz'
export PLACEHOLDER='placeholder'
```

**Review by artsyjian — APPROVED** [review]
amazing!

**Pull request merged**